UnityProject/Assets/Scripts/Objects/Atmospherics/AcuDevice.cs

```diff
@@ -0,0 +1,89 @@
+using UnityEngine;
+using Systems.Atmospherics;
+using Systems.ObjectConnection;
+
+namespace Objects.Atmospherics
+{
+	/// <summary>Allows an <seealso cref="AcuDevice"/> to be used for air quality sampling and control.</summary>
+	public interface IAcuControllable
+	{
+		/// <summary>The <c>GasMix</c> the device reports for sampling. Typically is for the device's tile.</summary>
+		GasMix AmbientGasMix { get; }
+
+		/// <summary>
+		/// The operating mode the controlling <see cref="AirController"/> has indicated the device should operate with.
+		/// <para>Each device type is responsible for interpreting their own behaviour from the <c>ACU</c>'s operating mode.</para>
+		/// </summary>
+		void SetOperatingMode(AcuMode mode);
+	}
+
+	/// <summary>
+	/// Allows an object with this component to be controlled by an <seealso cref="AirController"/>.
+	/// <para>When the object is initialised by the server, the referenced (mapped) <c>ACU</c> connects to the device.</para>
+	/// <para>See also <seealso cref="CustomInspectors.SlaveDeviceInspector"/>.</para>
+	/// </summary>
+	public class AcuDevice : MonoBehaviour, IServerLifecycle, IMultitoolSlaveable
+	{
+		/// <summary>The controller this device should link with at server initialisation.</summary>
+		public AirController Controller;
+
+		private IAcuControllable device;
+
+		private void Awake()
+		{
+			device = GetComponent<IAcuControllable>();
+			if (device == null)
+			{
+				Logger.LogError($"{this} has no component that implements {nameof(IAcuControllable)}!");
```

Comment:
```suggestion
				Logger.LogError($"{gameObject.Name} has no component that implements {nameof(IAcuControllable)}!");
```

Dont think `this` returns the object name?

Reply:
It returns the component and the gameobject. IIRC.

Reply:
ah